src/distro/distro.py

```diff
@@ -1078,6 +1082,47 @@ def uname_attr(self, attribute: str) -> str:
         """
         return self._uname_info.get(attribute, "")
 
+    def __resolve_chroot_symlink(self, link_location: str) -> str:
+        # only resolve symlink if root_dir is set and link is under it
+        if (
+            self.root_dir is None
+            or os.path.commonprefix([self.root_dir, link_location]) != self.root_dir
```

Comment:
The second half of this condition feels like we can either drop it or raise an exception if it happens. This silent version seems to either be trouble or not needed.

Reply:
I've added this condition as this method might be called with paths not in `root_dir`.
For instance with `LinuxDistribution(root_dir="/mnt", os_release_file="/etc/os-release")`, `os_release_file` is honored (as user-provided), and not derived from `root_dir` (see current version of the constructor).
Feel free to tell me that we should rename this method, or change its prototype.

I've also noticed we are calling `os.path.isfile` (which resolves symbolic links...), so I changed the patch to prevent/workaround this.

Reply:
> I've added this condition as this method might be called with paths not in `root_dir`. For instance with `LinuxDistribution(root_dir="/mnt", os_release_file="/etc/os-release")`, `os_release_file` is honored (as user-provided), and not derived from `root_dir` (see current version of the constructor). Feel free to tell me that we should rename this method, or change its prototype.

If the method can be called like that, then the result could be expected to be resolved while this new code would return plain `link_location` without any resolution. I think I'd vote for only calling `__resolve_chroot_symlink` when we have something from within the chroot. It would help the method do one thing only.


> I've also noticed we are calling `os.path.isfile` (which resolves symbolic links...), so I changed the patch to prevent/workaround this.

That's a great catch, but I'm not sure the fix works as intended. I'd think we needed a version of `os.path.isfile` that uses `__resolve_chroot_symlink` internally.

Reply:
> If the method can be called like that, then the result could be expected to be resolved while this new code would return plain `link_location` without any resolution. I think I'd vote for only calling `__resolve_chroot_symlink` when we have something from within the chroot. It would help the method do one thing only.

I started trying to implement this, it's even messier, as it would add many test conditions here and there... If you have an idea on your side, please feel free to edit this branch :roll_eyes:

> That's a great catch, but I'm not sure the fix works as intended. I'd think we needed a version of `os.path.isfile` that uses `__resolve_chroot_symlink` internally.

I couldn't get your point, sorry Sebastian :disappointed: